src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java

```diff
@@ -359,6 +371,12 @@ private <T, C extends GitSCMSourceContext<C, R>, R extends GitSCMSourceRequest>
                                                                                                  @CheckForNull Item retrieveContext,
                                                                                                  boolean delayFetch)
             throws IOException, InterruptedException {
+        if (!isFIPSCompliantTLS(this.getCredentialsId(), this.getRemote())) {
```

Comment:
similar lines must be added in method `retrieveActions` as it is used by mbp scan
```
hudson.plugins.git.GitException: Command "git ls-remote --symref -- http://foo.com/foo" returned status code 128:
stdout: 
stderr: remote: not found
fatal: repository 'http://foo.com/foo/' not found

	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:2846)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandWithCredentials(CliGitAPIImpl.java:2185)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandWithCredentials(CliGitAPIImpl.java:2079)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandWithCredentials(CliGitAPIImpl.java:2070)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.getRemoteSymbolicReferences(CliGitAPIImpl.java:3842)
	at jenkins.plugins.git.AbstractGitSCMSource.retrieveActions(AbstractGitSCMSource.java:1171)
	at jenkins.scm.api.SCMSource.fetchActions(SCMSource.java:847)
```
this one should have been some IAE exception